src/adapters/http/client.ts

```diff
@@ -0,0 +1,72 @@
+import Adapter from "../../lib/adapter.js"
+import GleeMessage from "../../lib/message.js"
+import got from "got"
+import { HttpAuthConfig, HttpAdapterConfig } from "../../lib/index.js"
+import http from "http"
+class HttpClientAdapter extends Adapter {
+
+  name(): string {
+    return "HTTP client"
+  }
+  async connect(): Promise<this> {
+    return this._connect()
+  }
+
+  async send(message: GleeMessage): Promise<void> {
+    return this._send(message)
+  }
+
+  async _connect(): Promise<this> {
+    this.emit("connect", {
+      name: this.name(),
+      adapter: this,
+      connection: http,
+      channel: this.channelNames,
+    })
+    return this
+  }
+  async _send(message: GleeMessage): Promise<void> {
+    const headers = {}
+    const config: HttpAdapterConfig = await this.resolveProtocolConfig("http")
+    const auth: HttpAuthConfig = await this.getAuthConfig(config.client.auth)
+    headers["Authentication"] = auth?.token
+    const serverUrl = this.serverUrlExpanded
+    this.channelNames.forEach(async (channelName) => {
+      const channelInfo = this.parsedAsyncAPI.channel(channelName)
+      const httpChannelBinding = channelInfo.binding("http")
+      const servers = channelInfo.servers()
+      if (
+        httpChannelBinding &&
+        (!servers.length || servers.includes(message.serverName))
+      ) {
+        const method = httpChannelBinding.method //get post put
+        const url = serverUrl + "/" + channelName
+        const body = message.payload.body
+        const query = message.payload.query
```

Comment:
You can use destructuring here:

```suggestion
        const { body, query } = message.payload
```

Also, please, assign the right type to these values.

Reply:
Also, you're getting the query parameters from the message payload? Why?

Reply:
there is no support for query params in the GleeMessage constructor, so i passed both the fields under payload only as message

Reply:
Let's explore where we put it, maybe in the GleeMessage object or maybe in the GleeConnection object. Definitely not in the payload as it's conditioning how we send and receive payloads. Where do you think it fits best?

Reply:
IMO `GleeMessage` should be the right place to add query as it already has other fields required for the message, so I have implemented a solution with that approach. Please have a look.